fix(targetTime): clarify timestamp formats and UTC interpretation

pchovelon opened the pull request:
Update the machinery dependency to v0.3.3 to ensure timestamps without timezone (both RFC3339 and PostgreSQL formats) are interpreted as UTC.

The documentation has been updated to clarify that targetTime accepts both RFC 3339 and PostgreSQL timestamp formats.

Closes #8928

Related: cloudnative-pg/machinery#167

pchovelon committed:
Fix documentation. Wrong targetTime format

Signed-off-by: Pierrick Chovelon <pierrick.chovelon@dalibo.com>

jsilvela reviewed (dismissed):
I think the issue with the targetTime in the document is the missing `Z` at the end.
The targetTime does admit RFC3339, RFC3339Micro, and timestamp.
That said, the code for handling is a bit convoluted and could use some updating.

pchovelon commented:
> Howdy @pchovelon . I've made some experiments.

Great, thank you @jsilvela . Running after time here ... I try to answer asap.

pchovelon reviewed (commented):
Change doc to include RFC3339 as the bug will be fixed with [#168](https://github.com/cloudnative-pg/machinery/pull/168)

pchovelon committed:
Apply suggestions from code review

Co-authored-by: Jaime Silvela <jaime.silvela@mailfence.com>
Signed-off-by: Pierrick <139142330+pchovelon@users.noreply.github.com>

mnencia committed:
fix(targetTime): treat RFC3339-like timestamps without timezone as UTC

Update the machinery dependency to include the fix that ensures
RFC3339-like timestamps without timezone (e.g., "2023-07-06T08:00:39")
are interpreted as UTC.

The documentation has been updated to clarify that RFC 3339 timestamps
without an explicit timezone suffix are interpreted as UTC.

Related: cloudnative-pg/machinery#167
Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

mnencia commented:
This PR has been updated to include the machinery dependency bump that ensures RFC3339-like timestamps without timezone (e.g., `2023-07-06T08:00:39`) are consistently interpreted as UTC.

Previously, there was an inconsistency where `ParseTargetTime` (used by barman-cloud for backup selection) interpreted them as UTC, while `ConvertToPostgresFormat` output them without timezone, letting PostgreSQL use server local time. This could cause backup selection and recovery to target different points in time if the server wasn't in UTC.

The documentation has also been updated to clarify this behavior.

Related PRs: cloudnative-pg/machinery#168 and cloudnative-pg/plugin-barman-cloud#700

armru committed:
chore: dep

Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

mnencia committed:
docs: clarify timestamp formats and UTC interpretation

Update documentation and API reference to clarify that targetTime accepts
both RFC3339 and PostgreSQL timestamp formats. Document that timestamps
without an explicit timezone are interpreted as UTC.

Add link to PostgreSQL documentation for recovery_target_time and include
a warning recommending users to always specify an explicit timezone to
avoid ambiguity.

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>
